formats/json/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt

```diff
@@ -186,6 +195,41 @@ public fun JsonArrayBuilder.addJsonObject(builderAction: JsonObjectBuilder.() ->
 public fun JsonArrayBuilder.addJsonArray(builderAction: JsonArrayBuilder.() -> Unit): Boolean =
     add(buildJsonArray(builderAction))
 
+/**
+ * Adds the given JSON [element] and [elements] to a resulting JSON array.
+ *
+ * @return `true` if the list was changed as the result of the operation.
+ */
+public fun JsonArrayBuilder.addAll(element: JsonElement, vararg elements: JsonElement): Boolean {
+    return addAll(listOf(element) + elements)
+}
+
+/** Adds the given string [values] to a resulting JSON array. */
+@JvmName("addAllStrings")
+public fun JsonArrayBuilder.addAll(values: Collection<String?>): Boolean =
+    addAll(values.map(::JsonPrimitive))
+
+/** Adds the given boolean [values] to a resulting JSON array. */
+@JvmName("addAllBooleans")
+public fun JsonArrayBuilder.addAll(values: Collection<Boolean?>): Boolean =
+    addAll(values.map(::JsonPrimitive))
+
+/** Adds the given numeric [values] to a resulting JSON array. */
+@JvmName("addAllNumbers")
+public fun JsonArrayBuilder.addAll(values: Collection<Number?>): Boolean =
+    addAll(values.map(::JsonPrimitive))
+
+/** Adds the given string [values] to a resulting JSON array. */
+public fun JsonArrayBuilder.addAll(vararg values: String?): Boolean =
+    addAll(values.toList())
+
+/** Adds the given boolean [values] to a resulting JSON array. */
+public fun JsonArrayBuilder.addAll(vararg values: Boolean?): Boolean =
+    addAll(values.toList())
+
+/** Adds the given numeric [values] to a resulting JSON array. */
+public fun JsonArrayBuilder.addAll(vararg values: Number?): Boolean =
+    addAll(values.toList())
 
```

Comment:
I considered adding a similar function to `add(null)`

https://github.com/Kotlin/kotlinx.serialization/blob/33104957873448189eb242413646b58883623e0a/formats/json/commonMain/src/kotlinx/serialization/json/JsonElementBuilders.kt#L117-L119

Something like 

```kotlin
@Suppress("UNUSED_PARAMETER") // allows to call `addAll(null)`
public fun JsonArrayBuilder.addAll(values: Collection<Nothing?>): Boolean = addAll(List(values.size) {JsonNull})
```

But I decided against it. It doesn't seem that useful. How often are there lists of type `Nothing`?

Interesting I discovered that Kotlin doesn't allow `vararg values: Nothing?`

<img width="306" alt="image" src="https://user-images.githubusercontent.com/897017/212500799-986b9e51-af5b-4680-b5ad-567fb44e9ac6.png">

so that's not an option.